hlog/internal/mutil/writer_proxy.go

```diff
@@ -59,7 +60,11 @@ func (b *basicWriter) WriteHeader(code int) {
 	if !b.wroteHeader {
 		b.code = code
 		b.wroteHeader = true
-		b.ResponseWriter.WriteHeader(code)
+		// if it's a websocket then we can't actually write the header
+		// on a hijacked session
+		if !b.isWs {
```

Comment:
Who exactly calls `WriteHeader` in question? Because probably this check should be made there and not inside `WriteHeader`.

Also, for hijacked connections `AccessHandler` has little sense and will not really count bytes going over the connection.

Reply:
Not sure if I follow? `hlog` wraps the original request, when the request is done, `hlog` closes the wrapped request/response and sets the status. It also tries to write a header with this status into the response. But this yields an error if the response has been hijacked (e.g. it is a websocket).

Reply:
I am trying to understand where this happens.

> closes the wrapped request/response

Where does it do that?

> It also tries to write a header with this status into the response.

Can you share a stack trace at the point this happens?

Reply:
You are right. I looked at this again, the error is actually within Gorilla websocket (e.g. https://github.com/gorilla/websocket/issues/852) -- not sure why I did see a (temporary) remediation of the issue by applying this patch. But it has shown again and it's not zerolog. It shows b/c I added the std log handler to zerolog in my program. Closing the PR. Thanks for pointing it out.

Reply:
I have recently used https://github.com/nhooyr/websocket in a stack with zerolog and it worked well for me.

Reply:
Thanks for mentioning this. I am using gorilla websockets quite extensively but maybe this is a worthy alternative. I was quite happy that they restarted the Gorilla suite eventually but with those latest changes things have gotten worse, not better (at least the parts that I am observing / need)